pipeline/container/routes/v4/runs.py

```diff
@@ -80,17 +85,96 @@ async def run(
         run_output = await response_queue.get()
 
         response_schema, response.status_code = _generate_run_result(run_output)
-        return response_schema
+
+        outputs = response_schema.outputs
+        if outputs is not None and any(
+            [output.type == run_schemas.RunIOType.stream for output in outputs]
+        ):
+            static_outputs = [
+                output
+                for output in outputs
+                if output.type != run_schemas.RunIOType.stream
+            ]
+
+            streaming_outputs = [
+                output
+                for output in outputs
+                if output.type == run_schemas.RunIOType.stream
+            ]
+
+            async def stream_func():
+                # Iterate over all streams, if done then add to static
+
+                streaming_outputs_to_move = []
+                # try:
+
+                while len(streaming_outputs) > 0:
+                    next_streaming_outputs = []
+                    for output in streaming_outputs:
+                        try:
+                            if output.value is None:
+                                raise Exception("Stream value was None")
+
+                            next_value = output.value.__next__()
+                            if next_value is not None:
+                                next_streaming_outputs.append(
+                                    run_schemas.RunOutput(
+                                        type=run_schemas.RunIOType.from_object(
+                                            next_value
+                                        ),
+                                        value=next_value,
+                                        file=None,
+                                    )
+                                )
+                        except StopIteration:
+                            static_outputs.append(
+                                run_schemas.RunOutput(
+                                    type=output.type,
+                                    value=None,
+                                    file=output.file,
+                                )
+                            )
+                            streaming_outputs_to_move.append(output)
+
+                    for output in streaming_outputs_to_move:
+                        streaming_outputs.remove(output)
```

Comment:
I don't quite understand what's going on here. From what I can see, it looks like if we reach the end of our stream we add a `None` to the final outputs in between the static outputs and the next streaming outputs?